contracts/LocalOracle.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+pragma solidity 0.7.1;
+
+import "./ERC1155ERC721.sol";
+import "./interfaces/ILocalOracle.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+
+contract LocalOracle is ILocalOracle {
+
+    using SafeMath for uint256;
+
+    // reference for ERC1155ERC721 contract
+    ERC1155ERC721 ERC1155;
+
+    // the event which will be emitted when help is needed from other blockchains
+    event HelpMe(uint256 indexed tokenId, address indexed owner);
+    // the event which will be emitted once help is received
+    event HelpReceived(uint256 indexed tokenId, address indexed owner, uint256 indexed amount);
+    // the event which will be emitted once a contract has burned their supply to help and proven 
+    // smart contract are indeed very generous
+    event ProvideHelp(uint256 indexed tokenId, address indexed owner, uint256 indexed amount);
+ 
+    // set the ERC1155 address
+    constructor(address _erc1155) {
+        ERC1155 = ERC1155ERC721(_erc1155);
+    }
+
+    /**
+        function which is callled when ERC115 quantity is burned
+     */
+    function helpMe(uint256 _quanityRemaining, address _account, uint256 _tokenId) external override {
+        if(_quanityRemaining < 2) {
```

Comment:
I would move this check to the calling `burn` function. Otherwise this external function could be unnecessarily
called anytime the quantity is not less than 2

Reply:
this is deliberately here and this function should decide what is considered "low inventory"

Reply:
OK.

Reply:
I agree with Ali, the token contract shall not implement the logic of knowing whether the inventory is low or not.
It is the role of the LocalOracle contract so it makes sense to have this here.